src/tests/test_raft_repl_dev.cpp

```diff
@@ -203,20 +204,20 @@ class TestReplicatedDB : public homestore::ReplDevListener {
             if (kv_snapshot_data.size() >= 1000) { break; }
         }
 
-        if (kv_snapshot_data.size() == 0) {
-            snp_data->is_last_obj = true;
-            LOGINFOMOD(replication, "Snapshot is_last_obj is true");
-            return 0;
+        if (kv_snapshot_data.size()) {
+            int64_t kv_snapshot_data_size = sizeof(KeyValuePair) * kv_snapshot_data.size();
+            sisl::io_blob_safe blob{static_cast< uint32_t >(kv_snapshot_data_size)};
+            std::memcpy(blob.bytes(), kv_snapshot_data.data(), kv_snapshot_data_size);
+            snp_data->blob = std::move(blob);
+            LOGINFOMOD(replication, "[Replica={}] Read logical snapshot callback obj_id={} term={} idx={} num_items={}",
+                       g_helper->replica_num(), snp_data->offset, s->get_last_log_term(), s->get_last_log_idx(),
+                       kv_snapshot_data.size());
+        } else {
+            LOGINFOMOD(replication, "no data need to be sent for snapshot");
         }
 
-        int64_t kv_snapshot_data_size = sizeof(KeyValuePair) * kv_snapshot_data.size();
-        sisl::io_blob_safe blob{static_cast< uint32_t >(kv_snapshot_data_size)};
-        std::memcpy(blob.bytes(), kv_snapshot_data.data(), kv_snapshot_data_size);
-        snp_data->blob = std::move(blob);
+        // we always read all the data in one shot, so we need to set is_last_obj to true
```

Comment:
I dont get this part, we limit each batch to 1000 in L204(```if (kv_snapshot_data.size() >= 1000) { break; }``` )

Reply:
yes , you are right, I miss this